### ISSUE TITLE
Rewrite silo import job

### ISSUE THREAD

**Issue by anna-parker** [loculus-project/loculus#2765]
Raised in https://github.com/loculus-project/loculus/pull/2764 - the silo_import job is very complex, instead we should
1. Refactor the import job into real code (potentially in python)
2. Create a container with silo_preprocessing and the import job

**Comment by corneliusroemer** [comment]
It'll be a bit annoying to create that container - unless there are SILO binaries readily available.

We'll have to see how easy it is to apt install stuff inside the existing silo container.

**Comment by theosanderson** [comment]
I think you (we) could have an additional container that just does the downloads and use a shared volume that it and Silo both access?

**Comment by corneliusroemer** [comment]
Yeah that's the other option I was thinking about - but then how do you trigger silo to start the processing? How do you monitor exit code etc? We want _one_ prepro container, adding a second makes things much more complicated again :/

**Comment by anna-parker** [comment]
silo preprocessing actually runs in the same container as lapissilo. The container includes python3 in the image: https://github.com/GenSpectrum/LAPIS-SILO/blob/main/Dockerfile - of course we might need further dependencies but we might just be able to use that container for now

**Comment by Taepper** [comment]
Actually python3 is only part of the container building the binary. The resulting container only contains [the silo binary](https://github.com/GenSpectrum/LAPIS-SILO/blob/c34fa1f18e8985b4f2b10cec57c28fd68fa76b02/Dockerfile#L18) and [the required programs](https://github.com/GenSpectrum/LAPIS-SILO/blob/c34fa1f18e8985b4f2b10cec57c28fd68fa76b02/Dockerfile#L20) for the healthcheck command

**Comment by corneliusroemer** [comment]
@Taepper you could potentially offer two images: one bare one and one based on standard Ubuntu.

That would make downstream uses easier, one can pick the minimal one required for the job.

**Comment by Taepper** [comment]
Let's quickly hop on a call? I think SILO can definitely offer that! But I am not quite sure on the specifics

**Comment by fhennig** [comment]
This topic came up again here: https://github.com/loculus-project/loculus/pull/3434

I think it makes sense to look again at migrating, possibly to Python.

- We build our own container image from scratch. We can build our image as a multi-stage build and use the SILO image as one stage, and copy the binary (andwhatever else we need) from there
- We add this image to the Pod
- We rewrite the shell script in Python

**Comment by theosanderson** [comment]
completed
